### ISSUE TITLE
Access issues when starting the Java Agent

### ISSUE THREAD

**Issue by tommskar** [microsoft/ApplicationInsights-Java#2031]
### Expected behavior
I expected logging from Logback to appear in Azure Application Insights.

### Actual behavior
According to `applicationinsights.log` the Java Agent starts just fine but quickly logs warnings/errors indefinitely.

### To Reproduce
1. Setup an Azure Web App along with Azure Application Insights
2. Stop the web app
3. Upload Java Agent to `/site/wwwroot/appinsights`
4. Add environment variable `JAVA_OPTS` and set it to `-javaagent:D:\home\site\wwwroot\appinsights\applicationinsights-agent-3.2.4.jar`
5. Add environment variable `APPLICATIONINSIGHTS_CONNECTION_STRING` and set it to your Application Insights key (`InstrumentationKey=...`).
6. Start the web app
7. Let it run for a bit
8. Stop the web app and inspect `applicationinsights.log` and Azure Application Insights.

#### Sample Application
N/A

### System information
- Java Version: 1.8.0_292
- Java Agent Version: 3.2.4
- OS type and version: Microsoft Windows NT 10.0.14393.0 (Azure Web App)
- Application Server type and version (if applicable): Tomcat 8.5.57
- Using spring-boot? No
- Additional relevant libraries (with version, if applicable): N/A

### Logs
applicationinsights.log
> 2022-01-04 23:33:59.063Z INFO  c.m.applicationinsights.agent - ApplicationInsights Java Agent 3.2.4 started successfully (PID 5472)
> 2022-01-04 23:34:49.281Z WARN  o.s.o.windows.WindowsOperatingSystem - Unable to open configured system Event log "System". Calculating boot time from uptime.
> 2022-01-04 23:34:50.110Z ERROR o.d.w.r.HkeyPerformanceDataUtil - Error reading performance data from registry for Process.
> 2022-01-04 23:34:52.329Z WARN  o.u.p.w.PerfCounterWildcardQuery - Disabling further attempts to query Process.
> 2022-01-04 23:34:53.860Z WARN  o.u.platform.windows.WmiQueryHandler - COM exception querying Win32_Process WHERE NOT Name LIKE"%_Total", which might not be on your system. Will not attempt to query it again. Error was -2147024891: Access is denied.(HRESULT: 80070005)
> 2022-01-04 23:34:55.407Z WARN  o.u.p.w.PerfCounterWildcardQuery - Disabling further attempts to query Processor Information.
> 2022-01-04 23:34:55.438Z WARN  o.u.platform.windows.WmiQueryHandler - COM exception querying Win32_PerfRawData_Counters_ProcessorInformation WHERE NOT Name LIKE "%_Total", which might not be on your system. Will not attempt to query it again. Error was -2147024891: Access is denied.(HRESULT: 80070005)
> 2022-01-04 23:35:48.894Z ERROR o.d.w.r.HkeyPerformanceDataUtil - Error reading performance data from registry for Process.
> 2022-01-04 23:35:50.066Z WARN  o.u.platform.windows.WmiQueryHandler - COM exception querying Win32_Process WHERE NOT Name LIKE"%_Total", which might not be on your system. Will not attempt to query it again. Error was -2147024891: Access is denied.(HRESULT: 80070005)
> 2022-01-04 23:35:50.863Z WARN  o.u.platform.windows.WmiQueryHandler - COM exception querying Win32_PerfRawData_Counters_ProcessorInformation WHERE NOT Name LIKE "%_Total", which might not be on your system. Will not attempt to query it again. Error was -2147024891: Access is denied.(HRESULT: 80070005)
> 2022-01-04 23:36:48.638Z ERROR o.d.w.r.HkeyPerformanceDataUtil - Error reading performance data from registry for Process.
> 2022-01-04 23:36:48.653Z WARN  o.u.platform.windows.WmiQueryHandler - COM exception querying Win32_Process WHERE NOT Name LIKE"%_Total", which might not be on your system. Will not attempt to query it again. Error was -2147024891: Access is denied.(HRESULT: 80070005)
> 2022-01-04 23:36:48.700Z WARN  o.u.platform.windows.WmiQueryHandler - COM exception querying Win32_PerfRawData_Counters_ProcessorInformation WHERE NOT Name LIKE "%_Total", which might not be on your system. Will not attempt to query it again. Error was -2147024891: Access is denied.(HRESULT: 80070005)

### Screenshots
N/A

### Addendum
This setup works fine locally and it did work fine in Azure when running version 2.x of the Application Insights Java Agent (although it bundled `com.microsoft.azure:applicationinsights-web-auto:2.6.1` and used `ApplicationInsights.xml` so maybe not entirely comparable). No `applicationinsights.json` was included.

**Comment by heyams** [comment]
Unfortunately, this is expected and a known issue. Azure App Service is in a sandbox environment. Certain shared resources like performance counters are not retrievable. When you run locally, it's a different env. 

How do you attach using 2.x on azure? I don't think web app works with the java 2.x sdk though.

**Comment by tommskar** [comment]
But shouldn't it still be able to pick up logging and such from Logback? I'm not really interested in performance counters so that's not a deal breaker for me but I assumed the errors above might be related to the missing log statements.

When I used Application Insights 2.x I had this `ApplicationInsights.xml` in the `resources` folder. I suspect the file is a boilerplate example from the Microsoft documentation.
```xml
<?xml version="1.0" encoding="utf-8"?>
<ApplicationInsights xmlns="http://schemas.microsoft.com/ApplicationInsights/2013/Settings" schemaVersion="2014-05-30">
    <!-- HTTP request component (not required for bare API) -->
    <TelemetryModules>
        <Add type="com.microsoft.applicationinsights.web.extensibility.modules.WebRequestTrackingTelemetryModule"/>
        <Add type="com.microsoft.applicationinsights.web.extensibility.modules.WebSessionTrackingTelemetryModule"/>
        <Add type="com.microsoft.applicationinsights.web.extensibility.modules.WebUserTrackingTelemetryModule"/>
    </TelemetryModules>

    <!-- Events correlation (not required for bare API) -->
    <!-- These initializers add context data to each event -->
    <TelemetryInitializers>
        <Add type="com.microsoft.applicationinsights.web.extensibility.initializers.WebOperationIdTelemetryInitializer"/>
        <Add type="com.microsoft.applicationinsights.web.extensibility.initializers.WebOperationNameTelemetryInitializer"/>
        <Add type="com.microsoft.applicationinsights.web.extensibility.initializers.WebSessionTelemetryInitializer"/>
        <Add type="com.microsoft.applicationinsights.web.extensibility.initializers.WebUserTelemetryInitializer"/>
        <Add type="com.microsoft.applicationinsights.web.extensibility.initializers.WebUserAgentTelemetryInitializer"/>
    </TelemetryInitializers>

    <PerformanceCounters>
        <Jmx>
            <Add objectName="java.lang:type=ClassLoading" attribute="TotalLoadedClassCount"
                 displayName="Loaded Class Count"/>
            <Add objectName="java.lang:type=Memory" attribute="HeapMemoryUsage.used"
                 displayName="Heap Memory Usage (used)" type="composite"/>
            <Add objectName="java.lang:type=Threading"
                 attribute="ThreadCount" displayName="Thread Count"/>
        </Jmx>
    </PerformanceCounters>
</ApplicationInsights>
```

Other than that I had the aforementioned `com.microsoft.azure:applicationinsights-web-auto:2.6.1` as an `implementation` dependency configuration in my `build.gradle` and the Java Agent was started as previously described.

**Comment by tommskar** [comment]
If the performance counter problems are known and tracked elsewhere we could just close this issue.

**Comment by trask** [comment]
hi @tommskar, can you give 3.2.5-BETA a try? we have tried to fix the log spam that you are seeing above, and would be great to know if that succeeded or not for your case. thx

**Comment by heyams** [comment]
@tommskar can you please try 3.2.5GA we just released yesterday? or the 3.2.5-BETA recommended by trask earlier? We fixed the oshi logging in beta. Thanks.

**Comment by trask** [comment]
@heyams the latest log above is from 3.2.5-BETA. it looks like the logging is coming directly from oshi, so we may need to tune the oshi logging levels

**Comment by tommskar** [comment]
@heyams: as note by @trask; I was using the 3.2.5-BETA.

**Comment by heyams** [comment]
@tommskar can you give this [snapshot](https://github.com/microsoft/ApplicationInsights-Java/suites/5186544192/artifacts/157757304) a try and let us know if that resolves your issue?

**Comment by heyams** [comment]
@tommskar i updated the snapshot link above.. earlier version was no good. thanks.

**Comment by heyams** [comment]
@tommskar Pinging... please let me know if this resolves your issue.  Once confirmed, i will include this fix in the new release. thanks.

**Comment by tommskar** [comment]
@heyams It works like a charm 😁 I let it run for about ten minutes without anything in the log file other than this:

> 2022-02-08 14:00:47.956Z INFO  c.m.applicationinsights.agent - ApplicationInsights Java Agent 3.2.6-BETA-SNAPSHOT started successfully (PID 21780)

**Comment by trask** [comment]
resolved by #2086
